exchanges/gateio/gateio.go

```diff
@@ -424,169 +949,2870 @@ func (g *Gateio) SendAuthenticatedHTTPRequest(ctx context.Context, ep exchange.U
+	params := url.Values{}
+	params.Set("underlying", underlying)
+	if contract != "" {
+		params.Set("contract", contract)
+	}
+	var response []ContractClosePosition
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodGet, optionsPositionClose, params, nil, &response)
+}
+
+// PlaceOptionOrder creates an options order
+func (g *Gateio) PlaceOptionOrder(ctx context.Context, arg OptionOrderParam) (*OptionOrderResponse, error) {
+	if arg.Contract == "" {
+		return nil, errInvalidOrMissingContractParam
+	}
+	if arg.OrderSize == 0 {
+		return nil, errInvalidOrderSize
+	}
+	if arg.Iceberg < 0 {
+		arg.Iceberg = 0
+	}
+	if arg.TimeInForce != gtcTIF && arg.TimeInForce != iocTIF && arg.TimeInForce != pocTIF {
+		arg.TimeInForce = ""
+	}
+	if arg.TimeInForce == iocTIF || arg.Price < 0 {
+		arg.Price = 0
+	}
+	var response OptionOrderResponse
+	return &response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, http.MethodPost,
+		optionsOrders, nil, &arg, &response)
+}
+
+// GetOptionFuturesOrders retrieves futures orders
+func (g *Gateio) GetOptionFuturesOrders(ctx context.Context, contract, underlying, status string, offset, limit uint64, from, to time.Time) ([]OptionOrderResponse, error) {
+	params := url.Values{}
+	if contract != "" {
+		params.Set("contract", contract)
+	}
+	if underlying != "" {
+		params.Set("underlying", underlying)
+	}
+	status = strings.ToLower(status)
+	if status == statusOpen || status == statusFinished {
+		params.Set("status", status)
+	}
+	if offset > 0 {
+		params.Set("offset", strconv.FormatUint(offset, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	if !from.IsZero() {
+		params.Set("from", strconv.FormatInt(from.Unix(), 10))
+	}
+	if !to.IsZero() && ((!from.IsZero() && to.After(from)) || to.Before(time.Now())) {
+		params.Set("to", strconv.FormatInt(to.Unix(), 10))
+	}
+	var response []OptionOrderResponse
+	return response, g.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot,
+		http.MethodGet, optionsOrders, params, nil, &response)
+}
+
+// CancelMultipleOptionOpenOrders cancels all open orders matched
+func (g *Gateio) CancelMultipleOptionOpenOrders(ctx context.Context, contract currency.Pair, underlying, side string) ([]OptionOrderResponse, error) {
+	params := url.Values{}
+	if !contract.IsEmpty() {
+		contract.Delimiter = currency.UnderscoreDelimiter
+		params.Set("contract", contract.String())
+	}
+	if underlying != "" {
+		params.Set("underlying", underlying)
+	}
+	if side != "" {
+		params.Set("side", side)
+	}
+	var response []OptionOrderResponse
+	return response, g.SendAuthenticatedHTTPRequest(ctx,
+		exchange.RestSpot, http.MethodDelete, optionsOrders, params, nil, &response)
+}
+
+// GetSingleOptionOrder retrieves a single option order
+func (g *Gateio) GetSingleOptionOrder(ctx context.Context, orderID string) (*OptionOrderResponse, error) {
+	if orderID == "" {
+		return nil, errInvalidOrderID
+	}
+	var order OptionOrderResponse
```

Comment:
```suggestion
	var o OptionOrderResponse
```
As `order` conflicts with package name